### PR TITLE
feat(frb_codegen): execute flutter pub get inside build_tool from cargokit on integrate and on create

### PR DESCRIPTION
**Pull request by MateusHBR**
## Changes
When setup a project, is necessary execute flutter pub get manually inside cargokit dart directory

## Checklist

- [ ] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to add a test.
- [x] `./frb_internal precommit --mode slow` (or `fast`) is run (it internal runs code generator, does auto formatting, etc).
- [ ] If this PR adds/changes features, documentations (in the `./website` folder) are updated.
- [ ] CI is passing. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to solve a failed CI.

## Remark for PR creator

- `./frb_internal --help` shows utilities for development.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

File: frb_codegen/src/library/commands/flutter.rs [hunk 1]
```diff
@@ -20,3 +21,9 @@ pub fn flutter_pub_add(items: &[String]) -> anyhow::Result<()> {
         *items
     )?)
 }
+
+#[allow(clippy::vec_init_then_push)]
+pub fn flutter_pub_get(path: &PathBuf) -> anyhow::Result<()> {
```
Comment (fzyzcjy) [inline frb_codegen/src/library/commands/flutter.rs:26]:
```suggestion
pub fn flutter_pub_get(path: &Path) -> anyhow::Result<()> {
```

(I am surprised CI does not warn this...)

**Review by fzyzcjy — APPROVED** [review]
Good job! Only a nit

**Comment by fzyzcjy** [comment]
For future readers:

1. Without this pub-get, projects can be run without problem
2. However, when opened via VSCode or other IDE, it will be red

image

**Review by fzyzcjy — COMMENTED** [review]
I tried the command, it says:

```
flutter pub get rust_builder/cargokit/build_tool 
  Using a naked argument for directory is deprecated and will stop working in a future Flutter release.

  Use --directory instead.
```

What about using `--directory`

File: frb_codegen/src/library/commands/flutter.rs [hunk 2]
```diff
@@ -20,3 +21,9 @@ pub fn flutter_pub_add(items: &[String]) -> anyhow::Result<()> {
...
+    info!("Execute `flutter pub get` inside {path:?}");
+    check_exit_code(&command_run!(call_shell[None, None], "flutter", "pub", "get", path)?)
```
Comment (fzyzcjy) [inline frb_codegen/src/library/commands/flutter.rs:28]: I tried the command, it says:

```
flutter pub get rust_builder/cargokit/build_tool 
  Using a naked argument for directory is deprecated and will stop working in a future Flutter release.

  Use --directory instead.
```

What about using `--directory`

Comment (MateusHBR) [reply frb_codegen/src/library/commands/flutter.rs:28]: @fzyzcjy I will fix it ASAP, I'm having some problems to execute ./frb_internal scripts, like precommit --mode slow

I don't know if it is related to my computer, I'm on a M1, but I'm investigating. The problem is related to ffigen, I don't know if you know how to solve, but sharing here.

Captura de Tela 2024-01-05 às 22 16 59

Comment (fzyzcjy) [reply frb_codegen/src/library/commands/flutter.rs:28]: @MateusHBR Firstly try to ignore it, as long as the script completes. There do exist some warnings on intel mac as well, which is unrelated to frb iirc.

Comment (MateusHBR) [reply frb_codegen/src/library/commands/flutter.rs]: @fzyzcjy Reading the macros I noticed that the execute could open a shell and execute directly inside a specific directory, But I am having problems to execute this script, adding some breaking points, the step of:

fn modify_permissions(dart_root: &Path) -> Result<()> {
    let dir_cargokit = dart_root.join("rust_builder").join("cargokit");

    #[cfg(not(windows))]
    {
        set_permission_executable(&dir_cargokit.join("build_pod.sh"))?;
        set_permission_executable(&dir_cargokit.join("run_build_tool.sh"))?;
        set_permission_executable(&dir_cargokit.join("run_build_tool.cmd"))?;
    }

    Ok(())
}

Is not founding the directory, do you know if I should add some additional step?

'No such file or directory (os error 2)'

Comment (fzyzcjy) [reply frb_codegen/src/library/commands/flutter.rs]: I am not sure what is the exact question... But you can mimic other usage code.

**Comment by MateusHBR** [comment]
@fzyzcjy Could you test?

tested here successfully :) !!!

**Comment by fzyzcjy** [comment]
Good job!

**Comment by fzyzcjy** [comment]
@all-contributors please add @MateusHBR for code

**Pull request merged**